### PR TITLE
Make Notification#user optional

### PR DESCRIPTION
**Pull request by alexdunae**
This is for Rails 5.x compatibility.

Also adds Appraisals for multi-Rails testing.

### CODE REVIEW

File: spec/notification_spec.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+describe Crier::Notification do
+  describe "Creation" do
+    let(:subject) { Subject.create!(title: "Subject") }
+
+    it "should create without a user" do
+      notification = Crier::Notification.create!(subject: subject)
+      expect(notification.valid?).to be_truthy
+    end
+
+    it "should create without a user" do
```
Comment (njakobsen) [inline spec/notification_spec.rb:12]:
```suggestion
    it "should be valid with a user" do
```

File: spec/notification_spec.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+require "spec_helper"
+
+describe Crier::Notification do
+  describe "Creation" do
```
Comment (njakobsen) [inline spec/notification_spec.rb:4]:
```suggestion
  describe "#valid?" do
```

File: spec/notification_spec.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    let(:subject) { Subject.create!(title: "Subject") }
+
+    it "should create without a user" do
```
Comment (njakobsen) [inline spec/notification_spec.rb:7]:
```suggestion
    it "should be valid without a user" do
```

File: lib/crier/notification.rb [hunk 4]
```diff
@@ -2,7 +2,7 @@ module Crier
   class Notification < ActiveRecord::Base
     self.table_name = 'crier_notifications'
 
-    belongs_to :crier, :class_name => 'User'
+    belongs_to :crier, :class_name => 'User', :optional => true
```
Comment (njakobsen) [inline lib/crier/notification.rb:5]: So this option ignored in older versions? It doesn't complain that it's never heard of it before?

**Review by njakobsen — CHANGES_REQUESTED** [review]

**Comment by alexdunae** [comment]
You caught this a bit before it was ready :) It should be all good now.  The `optional` does indeed cause an error in Rails 4 - the "appraisal" was incorrect before.

**Pull request merged**